cmd/internal/cli/push.go

```diff
@@ -34,6 +45,32 @@ var PushCmd = &cobra.Command{
 	Run: func(cmd *cobra.Command, args []string) {
 		// Push to library requires a valid authToken
 		if authToken != "" {
+			if !unauthenticatedPush {
+				// check if the container is signed
+				imageSigned, err := signing.IsSigned(args[0], "https://keys.sylabs.io", 0, false, authToken, true)
+				if err != nil {
+					sylog.Warningf("%v", err)
+				}
+				// if its not signed, print a warning
+				if !imageSigned {
+					sylog.Infof("TIP: Learn how to sign your own containers here : https://www.sylabs.io/guides/3.0/user-guide/signNverify.html")
+					fmt.Fprintf(os.Stderr, "\n")
+					sylog.Warningf("Your container is **NOT** signed! You REALLY should sign your container before pushing!")
+					fmt.Fprintf(os.Stderr, "Do you really want to continue? [N/y] ")
```

Comment:
Agree, but should we import `sypgp` just to ask a question? Should there be a different package for that?

Reply:
Since we seem to be asking questions in several places, I believe we should add a package for that. No need to do that in the context of this commit, though. This code can temporarily import sypgp and a future PR can refactor that into a separate package and have everyone use that code.